### PR TITLE
Switching to micromamba for install and add cluster docs

### PR DESCRIPTION
**Pull request by mikemhenry**

### CODE REVIEW

**Comment by mikemhenry** [comment]
I've asked the mamba people if this is a good idea or not: `export MAMBA_ROOT_PREFIX=$(conda config --show root_prefix | cut -d' ' -f2)` which would set the mamba root to be same as an existing conda install, which should share the package cache and envs. I am not sure if that does more harm than good, but I want to make sure our install guide is robust so that if someone has conda or mamba already installed, this won't break those envs.

**Comment by mikemhenry** [comment]
I am also switching the env name from `openfe` to `openfe_env` in this PR

File: docs/installation.rst [hunk 1]
```diff
@@ -14,26 +14,38 @@ Installing ``openfe``
 When you install ``openfe`` through any of the methods described below, you
 will install both the core library and the command line interface (CLI). 
 
-Installation with ``conda`` (recommended)
+Installation with ``micromamba`` (recommended)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-We recommend installing ``openfe`` with ``conda``, because it provides easy
+We recommend installing ``openfe`` with ``micromamba``, because it provides easy
```
Comment (IAlibay) [inline docs/installation.rst:20]: link to micromamba install instructions? (unless I've missed it somewhere)

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: docs/installation.rst (hunk 1 above)
Comment (mikemhenry) [reply docs/installation.rst:20]: Yes I just totally forgot!

**Comment by mikemhenry** [comment]
- [ ] Add cluster docs
- [ ] Add bit about cuda + conda interaction

File: docs/installation.rst [hunk 2]
```diff
@@ -14,26 +14,38 @@ Installing ``openfe``
...
-OpenMM and ambertools, which are needed by ``openfe``. 
-Conda can be installed either with the `full Anaconda distribution
-<https://www.anaconda.com/products/individual>`_, or with
-the `smaller-footprint miniconda
-<https://docs.conda.io/en/latest/miniconda.html>`_.
+OpenMM and ambertools, which are needed by ``openfe``.
 
-You can get ``openfe`` from the ``conda-forge`` channel. To install the most
-recent release in its own environment, use the commands ::
+To install ``micromamba`` execute the `installation script <https://mamba.readthedocs.io/en/latest/installation.html#install-script>`_.
```
Comment (richardjgowers) [inline docs/installation.rst:24]: Maybe a sentence here explaining that micromamba is conda on steroids.  People might have heard of conda and think it's what they should be using.

File: docs/installation.rst [hunk 3]
```diff
@@ -14,26 +14,38 @@ Installing ``openfe``
...
-  $ conda activate openfe
+On Linux the default shell is ``bash`` ::
+  
+  $ curl micro.mamba.pm/install.sh | bash
+
+And on macOS the default shell is ``zsh`` ::
+
+  $ curl micro.mamba.pm/install.sh | zsh
+
+Now we will set ``conda-forge`` as the default channel ::
```
Comment (richardjgowers) [inline docs/installation.rst:34]: "This channel will be the where software packages are downloaded from"

File: docs/installation.rst [hunk 4]
```diff
@@ -14,26 +14,38 @@ Installing ``openfe``
...
+
+  $ micromamba config append channels conda-forge
+
+To install ``openfe`` run ::
```
Comment (richardjgowers) [inline docs/installation.rst:38]: "This creates an environment called `openfe_env` which contains the `openfe` package, and all dependencies.

**Comment by mikemhenry** [comment]
Thanks for the feedback! I will be finishing this one off today!

**Comment by mikemhenry** [comment]
I updated the headings to follow this style https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

File: docs/installation.rst (hunk 2 above)
Comment (mikemhenry) [reply docs/installation.rst:24]: Done!

File: docs/installation.rst (hunk 1 above)
Comment (mikemhenry) [reply docs/installation.rst]: Done!

File: docs/installation.rst (hunk 4 above)
Comment (mikemhenry) [reply docs/installation.rst]: Done!

File: docs/guide/hpc.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                           https://conda.anaconda.org/conda-forge/noarch
+        base environment : /lila/home/henrym3/micromamba
+                platform : linux-64
+
+We can see that there is a virtual package ``__cuda=11.7=0``.
+This means that if we run a ``micromamba install`` command on a node with a GPU, the solver will install the correct version of the ``cudatoolkit``.
+However, if we ran the same command on the login node, the solver may install the wrong version of the ``cudatoolkit``, or depending on how the conda packages are setup, a CPU only version of the package.
+We can control the virtual package with the environmental variable ``CONDA_OVERRIDE_CUDA``.
+So on the login node, we can run ``CONDA_OVERRIDE_CUDA=11.3 micromamba info`` and see that the "correct" virtual CUDA is listed.
+For example, to install a version of ``openfe`` which is compatible with ``cudatoolkit 11.3`` run ``$ CONDA_OVERRIDE_CUDA=11.3 micromamba install openfe``.
```
Comment (IAlibay) [inline docs/guide/hpc.rst:77]: Is there meant to be a `$` before the `CONDA_OVERRIDE_CUDA` call?

File: docs/installation.rst [hunk 6]
```diff
@@ -14,31 +14,48 @@ Installing ``openfe``
...
+Now we need to activate our new environment ::
+
+  $ micromamba activate openfe_env
+
+To make sure everything is working, run the tests ::
 
-That creates a separate environment for ``openfe`` that won't interfere with
-other things you have installed. You will need to activate the ``openfe``
-environment before using it in a new shell.
+  $ pytest --pyargs openfe
```
Comment (IAlibay) [inline docs/installation.rst:49]:
```suggestion
  $ pytest --pyargs openfe openfecli
```
? (can't remember if you also need a second pyargs before openfecli

File: docs/installation.rst [hunk 7]
```diff
@@ -227,10 +244,18 @@ Now the CLI tool should work as well ::
                file
      quickrun  Run a given transformation, saved as a JSON file
 
- 
+To make sure everything is working, run the tests ::
+
+  $ pytest --pyargs openfe
```
Comment (IAlibay) [inline docs/installation.rst:249]: as above

File: docs/installation.rst [hunk 8]
```diff
@@ -270,16 +295,26 @@ You can access the ``openfe`` CLI from the singularity image with ::
 
   $ singularity run --nv openfe_0.7.4-apptainer.sif openfe --help
 
+To make sure everything is working, run the tests ::
+
+  $ singularity run --nv openfe_0.7.4-apptainer.sif pytest --pyargs openfe
```
Comment (IAlibay) [inline docs/installation.rst:300]: as above

**Review by IAlibay — COMMENTED** [review]
couple of quick comments

File: docs/guide/hpc.rst (hunk 5 above)
Comment (mikemhenry) [reply docs/guide/hpc.rst]: Ah no, it is a copy and paste artifact from when I had it formatted like a bash prompt, good catch! Fixed

File: docs/installation.rst (hunk 6 above)
Comment (mikemhenry) [reply docs/installation.rst]: Just tested it and you don't need a second pyargs

Comment (mikemhenry) [reply docs/installation.rst]: done!

File: docs/installation.rst (hunk 7 above)
Comment (mikemhenry) [reply docs/installation.rst]: done!

File: docs/installation.rst (hunk 8 above)
Comment (mikemhenry) [reply docs/installation.rst]: done!

**Comment by mikemhenry** [comment]
Hmm, I did get this error when I ran 
```
ERROR openfecli/tests/test_cli.py::TestCLI::test_command_sections - UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 3173: ordinal not in range(128)
ERROR openfecli/tests/test_cli.py::TestCLI::test_get_installed_plugins - UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 3173: ordinal not in range(128)
```

`pytest --pyargs openfe openfecli` but I don't get that error when I run it separately, do we have another fixutre fun time? Can anyone reproduce?

File: docs/installation.rst (hunk 6 above)
Comment (richardjgowers) [reply docs/installation.rst]: This'll just turn into the OpenFE test command anyway

**Comment by richardjgowers** [comment]
@mikemhenry Yeah I hit that on #373

**Comment by mikemhenry** [comment]
Hmm, I wonder if it has to do with running the test in the root of the repo, I will test that.

**Comment by mikemhenry** [comment]
Yes, not running in the root dir of the repo fixed it

**Review by richardjgowers — APPROVED** [review]

**Review by IAlibay — APPROVED** [review]
sorry didn't realise I had left a blocking review - lgtm

**Pull request merged**